src/components/Item/VisualizationItem/Visualization/DefaultPlugin.js

```diff
@@ -1,111 +1,64 @@
+import { load, unmount } from './plugin'
 import { getGridItemDomId } from '../../../ItemGrid/gridUtil'
 
-const pluginCredentials = d2 => {
-    return {
-        baseUrl: getBaseUrl(d2),
+const DefaultPlugin = ({ item, activeType, visualization, options, style }) => {
+    const { d2 } = useD2({})
+    const { baseUrl } = useConfig()
+    const credentials = {
+        baseUrl,
         auth: d2.Api.getApi().defaultHeaders.Authorization,
     }
-}
-
-class DefaultPlugin extends Component {
-    pluginCredentials = null
-
-    constructor(props, context) {
-        super(props)
-
-        this.d2 = context.d2
-    }
 
-    pluginIsAvailable = () =>
-        pluginManager.pluginIsAvailable(
-            this.props.activeType || this.props.item.type
-        )
+    const prevItem = useRef()
+    const prevActiveType = useRef()
 
-    shouldPluginReload = prevProps => {
-        // TODO - fix this hack, to handle bug with multiple
-        // rerendering while switching between dashboards.
-        //
-        // To determine if the rendering is happening because of a
-        // dashboard switch, check if the item reference has changed.
-        const reloadAllowed = this.props.item === prevProps.item
+    useEffect(() => {
+        load(item, visualization, {
+            credentials,
+            activeType,
+            options,
+        })
 
-        const filtersChanged = prevProps.itemFilters !== this.props.itemFilters
-        const vis = orObject(this.props.visualization)
-        const prevVis = orObject(prevProps.visualization)
-        const visChanged =
-            vis.id !== prevVis.id ||
-            prevProps.activeType !== this.props.activeType
+        prevItem.current = item
+        prevActiveType.current = activeType
 
-        return reloadAllowed && (visChanged || filtersChanged)
-    }
+        return () => unmount(item, activeType)
+    }, [])
 
-    reloadPlugin = prevProps => {
-        if (this.pluginIsAvailable() && this.shouldPluginReload(prevProps)) {
-            if (
-                this.props.activeType !== prevProps.activeType ||
-                this.props.itemFilters !== prevProps.itemFilters
-            ) {
-                pluginManager.unmount(this.props.item, prevProps.activeType)
-
-                pluginManager.load(this.props.item, this.props.visualization, {
-                    credentials: this.pluginCredentials,
-                    activeType: this.props.activeType,
-                })
-            }
-        }
-    }
-
-    componentDidMount() {
-        this.pluginCredentials = pluginCredentials(this.d2)
-
-        if (this.pluginIsAvailable()) {
-            pluginManager.load(this.props.item, this.props.visualization, {
-                credentials: this.pluginCredentials,
-                activeType: this.props.activeType,
-                options: this.props.options,
+    useEffect(() => {
+        if (shouldPluginReload()) {
+            unmount(item, prevActiveType.current)
```

Comment:
If I understand correctly `prevItem.current` will always be the same as `item`, so it doesn't matter which one is used here.
Correct?

Reply:
Right. So the goal here is that if the user changes the active type (i.e., EV -> ER), then the EV plugin should unmount, and the ER plugin should load.

Reply:
👍